jupyterhub/apihandlers/users.py

```diff
@@ -150,6 +150,12 @@ def patch(self, name):
             raise web.HTTPError(404)
         data = self.get_json_body()
         self._check_user_model(data)
+        # check if the new name is already taken
+        new_name = self.find_user(data['name']) if 'name' in data else None
+        if new_name is not None:
```

Comment:
but basically if you patch /hub/api/user/chris with this body

{
'name': 'chris',
'admin': true
}

you will get User %s already exists, username must be unique